ui/src/main/java/edu/wpi/grip/ui/preview/PreviewsView.java

```diff
@@ -48,24 +58,196 @@ public PreviewsView(EventBus eventBus) {
     }
 
     @Subscribe
+    /**
+     * This function is called when a step moves in the pipeline to adjust the positions of any open previews it has
+     * to reflect the new order of the pipeline.
+     */
+    public synchronized void onPreviewOrderChanged(StepMovedEvent event) {
```

Comment:
To be honest, I'd rather just leave it as is and come back to this later- at this point, logging seems more important to me. Is this something that you think is critical before this can be merged? I can always come back to this later. But I do think you're right about this method being cleaner. I avoided this in this instance because I forgot about lambdas (I learned java without them). At the very least, next time I will be sure to use this method. Thank you for bringing it up :)